### DISCUSSION TITLE
2.24 (Draft) - Database release

### DISCUSSION THREAD

**Discussion by Geektoolkit** [Geektoolkit/Dynaframe3#135] [General]
This is a draft (highly untested) release. Please do not install unless curious.
Frame syncing is broken in this release.
Video playback MAY be broken currently.
Startup time (first run) is VERY long currently. 

This is the first release of testing moving things to a sql database. I fixed a few of the UI bugs to bring back things like screen on/off that broke in 2.23.  This is a test release to see if I can install from the web while I work on the first run performance and a few other things.  If you do bravely test this please leave me feedback on how it goes. 

<hr /><em>This discussion was created from the release <a href='https://github.com/Geektoolkit/Dynaframe3/releases/tag/2.24'>2.24 (Draft) - Database release</a>.</em>

**Comment by RUHavingFun** [comment]
So far looks pretty good.  Does take a while to start like you said.  Seems some playback items are off.  Would like to see if you set playback to honor the duration and it is a shorter video to have it loop.  I am also seeing an image between transitions.

https://user-images.githubusercontent.com/39495965/127935805-4bd1b40f-6239-41c6-a76c-cb0db1f943ee.mp4


Uploading IMG_9430.MOV…

**Comment by qwksilver** [comment]
it used to have the oldest image as a background when switching over to video it would keep that image up, the behaviour changed around 2.18ish to use a blank screen image instead, since 2.24 was a big rewrite it may be that transition blanking is part of what's broken, it will likely get cleared up in a soon release.  as for more controls on video playback and internal looping to match playtime, well the playlist doesn't really know the durations, so it doesn't make those choices.  most of us are just curating our videos  I think there is a tutorial on building smooth loops with simple ffmpeg commands.

**Reply by RUHavingFun** [reply]
Would it have to know playtime?  Could it not just loop video and then transition when the time is up?

**Reply by Geektoolkit** [reply]
The video playback is a balance of trying to not have too many options and make it confusing vs. trying to solve for every scenario.  There's a balance between users managing the content (to have the timing consistent) vs. trying to have the app be smart about it.  Right now there's two settings "obey transition time' and 'obey video length time'.  If the video is 2 mins on a 30 second transition, then the first option transitions at 30 seconds, forcing the video to stop. the second option transitions at 2 mins.  
  Now what if there is a video that is 15 seconds on a 30 second transition time? Currently I force a transition at 15 seconds to keep from having a blank background for yet another 15.  I may put in yet another option to loop to transition, but it's not there yet.  
   When I designed/used this I intended the video functions to be things like cinemagraphs or looping animations that would be easy to curate to the transition time. I have seen people play everything from movie trailers to family videos since then and so it's a matter of catching up to how others are using it and designing the right solutions from there. But most of my work right now has been on polishing up features like comitup and providing an image. I should have an announcement of where the projects heading beyond this soon.
